File: SingletonDemo/Singleton.js
const fs = require("node:fs");

class Logger {

  logFile;

  constructor() {
    this.logFile = fs.openSync("app.log", "w");
  }

  info(message) {
    fs.writeSync(this.logFile, `INFO: ${message}`);
  }

  warn(message) {
    fs.writeSync(this.logFile, `WARN: ${message}`);

  }

  error(message) {
    fs.writeSync(this.logFile, `ERROR: ${message}`);

  }

  close() {
    console.log(this.logFile);
  }
}

const log = new Logger();
log.warn("This is a warning");
log.close();

File: SingletonDemo/Singleton.py

class Logger:
  def __init__(self):
    self.log_file = open("app.log", "w")

  def info(self, message: str):
    self.log_file.write(f"INFO: {message}\n")

  def warn(self, message: str):
    self.log_file.write(f"WARN: {message}\n")

  def error(self, message: str):
    self.log_file.write(f"ERROR: {message}\n")

  def close(self):
    self.log_file.close()

log=Logger()
log.warn("This is a warning")
log.close()